pkg/common/providers/ocmprovider/cluster.go

```diff
@@ -518,7 +518,13 @@ func (o *OCMProvider) ClusterKubeconfig(clusterID string) ([]byte, error) {
 		return getLocalKubeConfig(localKubeConfig)
 	}
 
+	existingKubeConfig := viper.GetString(config.Kubeconfig.Contents)
+	if existingKubeConfig != "" {
+		return []byte(existingKubeConfig), nil
+	}
+
 	if creds, ok := o.credentialCache[clusterID]; ok {
+		viper.Set(config.Kubeconfig.Contents, creds)
```

Comment:
How does this reduce calls to one? do you store the kubeconfig for every cluster?